homeassistant/components/camera/synology.py

```diff
@@ -0,0 +1,131 @@
+DEFAULT_STREAM_ID = '0'
+TIMEOUT = 10
+
+# pylint: disable=line-too-long
+MJPEG_URL = '{0}/webapi/SurveillanceStation/streaming.cgi?api=\
+SYNO.SurveillanceStation.Streaming&method=LiveStream&version=1&cameraId={1}'
+STILL_IMAGE_URL = '{0}{1}'
+LOGIN_URL = '{0}/webapi/auth.cgi?api=SYNO.API.Auth&method=\
+Login&version=2&account={1}&passwd={2}&session=SurveillanceStation&format=sid'
+URL = '{0}/webapi/entry.cgi?\
+api=SYNO.SurveillanceStation.Camera&method=List&version=1'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_URL): cv.string,
+    vol.Required(CONF_CAMERA_NAME): cv.string,
+    vol.Optional(CONF_STREAM_ID, default=DEFAULT_STREAM_ID): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup a Synology IP Camera."""
+    add_devices([SynologyCamera(config)])
+
+
+# pylint: disable=too-many-instance-attributes
+class SynologyCamera(Camera):
+    """An implementation of a Synology NAS based IP camera."""
+
+    def __init__(self, config):
+        """Initialize a Synology Surveillance Station camera."""
+        from json import loads
+        super().__init__()
+        self._name = config.get(CONF_NAME)
+        self._username = config.get(CONF_USERNAME)
+        self._password = config.get(CONF_PASSWORD)
+        self._synology_url = config.get(CONF_URL)
+        self._camera_name = config.get(CONF_CAMERA_NAME)
+        self._stream_id = config.get(CONF_STREAM_ID)
+
+        self.get_sid()
+
+        # With our session id (sid) we can get the snapshot path.
+        sidurl = URL.format(self._synology_url)
+        req = requests.get(
+            sidurl, timeout=TIMEOUT, verify=False, cookies={'id': self._sid})
+        cam_resp = loads(req.text)
+        cameras = cam_resp['data']['cameras']
+
+        for camera in cameras:
+            if camera['name'] == self._camera_name:
+                snapshot_path = camera['snapshot_path']
+                self._cam_id = str(camera['id'])
+                self._still_image_url = STILL_IMAGE_URL.format(
+                    self._synology_url, snapshot_path)
+                self._mjpeg_url = MJPEG_URL.format(
+                    self._synology_url, self._cam_id)
+
+    def get_sid(self):
+        """Get a session id."""
+        from json import loads
+        login_url = LOGIN_URL.format(
+            self._synology_url, self._username, self._password)
+        req = requests.get(login_url, timeout=TIMEOUT, verify=False)
+        sid_resp = loads(req.text)
+        sids = sid_resp['data']
+        self._sid = sids['sid']
+
+    def camera_image(self):
+        """Return a still image response from the camera."""
+        try:
+            response = requests.get(
+                self._still_image_url, timeout=TIMEOUT,
+                verify=False, cookies={'id': self._sid})

```

Comment:
It still prevent MITM attacks, but you're still vulnerable to an impersonation attack.  I'd still prefer it to sending username/password in clear text.  I'll change it to a configurable option, with the default assumption that people have a valid cert.  That will be more secure, but will probably result in more problems for people who don't understand the options or how certificates work.